File: 프로그래머스/unrated/181871. 문자열이 몇 번 등장하는지 세기/문자열이 몇 번 등장하는지 세기.cs
using System;

public class Solution {
    public int solution(string myString, string pat) 
    {
        int answer = 0;
        int lastIndex = 0;
        
        while(true)
        {
            lastIndex = myString.IndexOf(pat, lastIndex);
            if(lastIndex == -1)
                break;
            
            ++lastIndex;
            ++answer;
        }
        
        return answer;
    }
}

File: 프로그래머스/0/181871. 문자열이 몇 번 등장하는지 세기/문자열이 몇 번 등장하는지 세기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string myString, string pat) {
    int answer = 0;
    size_t pos = 0;
    
    while (true)
    {
        pos = myString.find(pat, pos);
        
        if (pos == std::string::npos)
            break;
        
        ++answer;
        ++pos;
    }
    
    return answer;
}